internal/provider/job_resource.go

```diff
@@ -0,0 +1,315 @@
+				Required: true,
+			},
+			"inventory_id": schema.Int64Attribute{
+				Optional: true,
+			},
+			"job_type": schema.StringAttribute{
+				Computed: true,
+			},
+			"job_url": schema.StringAttribute{
+				Computed: true,
+			},
+			"status": schema.StringAttribute{
+				Computed: true,
+			},
+			"extra_vars": schema.StringAttribute{
+				Optional:   true,
+				CustomType: jsontypes.NormalizedType{},
+			},
+			"triggers": schema.MapAttribute{
+				Optional:    true,
+				ElementType: types.StringType,
+				Description: "A map of arbitrary key/values to be set into the HTTP data request when creating/updating the Job resource.",
+			},
+			"ignored_fields": schema.ListAttribute{
+				ElementType: types.StringType,
+				Computed:    true,
+				Description: "The list of properties set by the user but ignored on server side.",
+			},
+		},
+	}
+}
+
+// jobResourceModel maps the resource schema data.
+type jobResourceModel struct {
+	TemplateID    types.Int64          `tfsdk:"job_template_id"`
+	Type          types.String         `tfsdk:"job_type"`
+	URL           types.String         `tfsdk:"job_url"`
+	Status        types.String         `tfsdk:"status"`
+	InventoryID   types.Int64          `tfsdk:"inventory_id"`
+	ExtraVars     jsontypes.Normalized `tfsdk:"extra_vars"`
+	IgnoredFields types.List           `tfsdk:"ignored_fields"`
+	Triggers      types.Map            `tfsdk:"triggers"`
+}
+
+var keyMapping = map[string]string{
+	"inventory": "inventory",
+}
+
+func (d *jobResourceModel) GetTemplateID() string {
+	return d.TemplateID.String()
+}
+
+func (d *jobResourceModel) GetURL() string {
+	if !d.URL.IsNull() && !d.URL.IsUnknown() {
+		return d.URL.ValueString()
+	}
+	return ""
+}
+
+func (d *jobResourceModel) ParseHTTPResponse(body []byte) error {
+	/* Unmarshal the json string */
+	var result map[string]interface{}
+	err := json.Unmarshal(body, &result)
+	if err != nil {
+		return err
+	}
+
+	d.Type = types.StringValue(result["job_type"].(string))
+	d.URL = types.StringValue(result["url"].(string))
+	d.Status = types.StringValue(result["status"].(string))
+	d.IgnoredFields = types.ListNull(types.StringType)
+
+	if value, ok := result["ignored_fields"]; ok {
+		var keysList = []attr.Value{}
+		for k := range value.(map[string]interface{}) {
+			key := k
+			if v, ok := keyMapping[k]; ok {
+				key = v
+			}
+			keysList = append(keysList, types.StringValue(key))
+		}
+		if len(keysList) > 0 {
+			d.IgnoredFields, _ = types.ListValue(types.StringType, keysList)
+		}
+	}
+
+	return nil
+}
+
+func IsValueProvided(value attr.Value) bool {
+	return !value.IsNull() && !value.IsUnknown()
+}
+
+func (d *jobResourceModel) CreateRequestBody() ([]byte, diag.Diagnostics) {
+	body := make(map[string]interface{})
+	var diags diag.Diagnostics
+
+	// Triggers
+	if IsValueProvided(d.Triggers) {
+		// Set manual triggers
+		for key, value := range d.Triggers.Elements() {
+			body[key] = value.(types.String).ValueString()
+		}
+	}
```

Comment:
Triggers should not be added to the HTTP request. They are just there as a way to force TF to update the resource. The reason this is needed is because of the nature of the job resource. A user may often want to launch a new job using the same data. TF is only going to do that if something has changed with the job resource definition, though. It's possible to do this with `terraform taint`, but the `triggers` attribute is being included so that the user can do the same by tying the new launch to whatever TF values they want.

Reply:
@gravesm it looks strange the way we want to use this field however I will update accordingly.
Is it ok if the field is defined as a string?

Reply:
It should be a map. See https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_invocation#triggers for an example of the pattern in use elsewhere in the TF ecosystem.

Reply:
updated